### ISSUE TITLE
'Unknown element' while using Ivy library compiled with compilationMode partial

### ISSUE THREAD

**Issue by bashoogzaad** [angular/vscode-ng-language-service#1538]
Hello there,

I have created an Angular Ivy library with compilationMode partial, which is recommended by the Angular team. I have published this privately on NPM and installed in my project. The project compiles fine, however, the templates keeps showing this error:

`'my-custom-component' is not a known element:`

The Angular Language Service is in View Engine Mode, but I do not seem to get it in Ivy mode... How does the service decide which mode to use? And how can this problem be solved? It is not stopping me from developing the project, but all the templates are completely red with errors...

ALS version: 12.2.2
Angular version: 12

Any help is greatly appreciated!

**Comment by atscott** [comment]
Hi @bashoogzaad do you have an example project? The Angular Language Service uses View Engine if any project in the workspace is using a version less than v9 or if you have the Legacy View Engine option enabled in the extension options.

**Comment by bashoogzaad** [comment]
Excuse me for reopening this, but I have found the issue. I did not read your answer well: _'if any project in the workspace is using a version less than v9'_ I indeed had one old project on Angular v8. After deleting this project from the workspace, the Ivy language service was enabled. 

Maybe I am the only one who thinks this is a bit strange, because the logs state it is reading the `tsconfig` file for the current project, so I do not understand what other projects in the workspace have to do with this. But I am sure there is a good reason for this!

Thanks for the effort and maybe this comment can be useful for anyone in the future. This can be closed now.
